service.subtitles.napisy24pl/resources/lib/NapisyUtils.py

```diff
@@ -0,0 +1,349 @@
+            item["tvshow"] = tvshow[0]
+        else:
+            item["tvshow"] = ''.join(tvshow)
+    else:
+        item["tvshow"] = tvshow[0]
+
+    item["title"] = unicode(item["title"], "utf-8")
+    item["tvshow"] = unicode(item["tvshow"], "utf-8")
+    # Removes country identifier at the end
+    item["title"] = re.sub(r'\([^\)]+\)\W*$', '', item["title"]).strip()
+    item["tvshow"] = re.sub(r'\([^\)]+\)\W*$', '', item["tvshow"]).strip()
+
+
+def parse_rls_title(item):
+    title = regexHelper.sub(' ', item["title"])
+    tvshow = regexHelper.sub(' ', item["tvshow"])
+
+    groups = re.findall(r"(.*?) (\d{4})? ?(?:s|season|)(\d{1,2})(?:e|episode|x|\n)(\d{1,2})", title, re.I)
+
+    if len(groups) == 0:
+        groups = re.findall(r"(.*?) (\d{4})? ?(?:s|season|)(\d{1,2})(?:e|episode|x|\n)(\d{1,2})", tvshow, re.I)
+
+    if len(groups) > 0 and len(groups[0]) >= 3:
+        title, year, season, episode = groups[0]
+        item["year"] = str(int(year)) if len(year) == 4 else year
+
+        item["tvshow"] = regexHelper.sub(' ', title).strip()
+        item["season"] = str(int(season))
+        item["episode"] = str(int(episode))
+        log("TV Parsed Item: %s" % (item,))
+
+    else:
+        groups = re.findall(r"(.*?)(\d{4})", item["title"], re.I)
+        if len(groups) > 0 and len(groups[0]) >= 1:
+            title = groups[0][0]
+            item["title"] = regexHelper.sub(' ', title).strip()
+            item["year"] = groups[0][1] if len(groups[0]) == 2 else item["year"]
+
+            log("MOVIE Parsed Item: %s" % (item,))
+
+
+class NapisyHelper:
+    BASE_URL = "http://napisy24.pl"
+
+    def __init__(self):
+        self.urlHandler = URLHandler()
+
+    def get_subtitle_list(self, item):
+        if item["tvshow"]:
+            search_results = self._search_tvshow(item)
+        else:
+            search_results = self._search_movie(item)
+
+        log("Search results: %s" % search_results)
+        results = self._build_subtitle_list(search_results, item)
+        log("Results: %s" % results)
+
+        return results
+
+    def download(self, id, zip_filename):
+        ## Cleanup temp dir, we recomend you download/unzip your subs in temp folder and
+        ## pass that to XBMC to copy and activate
+        if xbmcvfs.exists(__temp__):
+            shutil.rmtree(__temp__)
+        xbmcvfs.mkdirs(__temp__)
+
+        subtitle_type_map = ["sru", "sr", "tmp", "mdvd", "mpl2"]
+        subs_format = int(__addon__.getSetting("subs_format"))
+
+        query = {"napisId": id,
+                 "typ": subtitle_type_map[subs_format]}
+
+        f = self.urlHandler.request(self.BASE_URL + "/download", query, referer=self.BASE_URL)
+
+        if f is None and self.login(True):
+            f = self.urlHandler.request(self.BASE_URL + "/download", query, referer=self.BASE_URL)
+
+        with open(zip_filename, "wb") as subFile:
+            subFile.write(f)
+        subFile.close()
+        xbmc.sleep(500)
```

Comment:
Please use something like `xbmc.Monitor().waitForAbort(0.5)` instead.
We discourage the usage of `xbmc.sleep`